### PR TITLE
Added check for bocked or pending domains. Fixes #1850

### LINKED ISSUE #1850: Check if links already exist when adding them to books
While a domain is still pending, people may add links multiple times because they aren't shown on the page. It would be helpful to let the user know that the link is already added and pending approval (or blocked)

### PR DESCRIPTION
**Pull request by willhoh**
added domain pending or blocked check

### CODE REVIEW

File: bookwyrm/forms.py [hunk 1]
```diff
@@ -226,6 +227,18 @@ class FileLinkForm(CustomForm):
...
+        """make sure the domain isn't blocked or pending"""
+        cleaned_data = super().clean()
+        url = cleaned_data.get('url')
+        domain = urlparse(url).netloc
+        if models.LinkDomain.objects.filter(domain=domain).exists():
+            status = models.LinkDomain.objects.get(domain=domain).status
+            if status == 'blocked':
+                self.add_error("url", _("Domain is blocked. Don't try this url again."))
+            elif status == 'pending':
+                self.add_error("url", _("Domain already pending. Please try later."))
```
Comment (mouse-reeve) [inline bookwyrm/forms.py:241]: It's actually just fine if users add links where the domain is pending -- once the domain is reviewed, the links added while the domain was pending will be shown on the book page. So this check can be removed.

File: locale/de_DE/LC_MESSAGES/django.po [hunk 2]
```diff
@@ -284,7 +292,7 @@ msgstr "Português Europeu (Portugiesisch)"
 
 #: bookwyrm/settings.py:258
 msgid "Swedish (Svenska)"
-msgstr ""
+msgstr "Swedish (Schwedisch)"
```
Comment (cincodenada) [inline locale/de_DE/LC_MESSAGES/django.po:295]: Not related to the main part of the PR, but this called it to my attention: shouldn't both of these be the other way around? All the other languages are "Native Spelling (Translated Spelling)", so I think these should be "Svenska (Swedish)" and "Svenska (Schwedish)"

File: bookwyrm/forms.py [hunk 3]
```diff
@@ -226,6 +227,18 @@ class FileLinkForm(CustomForm):
...
+    
+    def clean(self):
+        """make sure the domain isn't blocked or pending"""
+        cleaned_data = super().clean()
+        url = cleaned_data.get('url')
+        domain = urlparse(url).netloc
+        if models.LinkDomain.objects.filter(domain=domain).exists():
+            status = models.LinkDomain.objects.get(domain=domain).status
+            if status == 'blocked':
+                self.add_error("url", _("Domain is blocked. Don't try this url again."))
```
Comment (mouse-reeve) [inline bookwyrm/forms.py:239]: This looks good, I would just tweak the wording a bit so it reads more gently. Maybe something like "This domain is blocked. Please contact your administrator if you think this is an error." ?

**Review by mouse-reeve — CHANGES_REQUESTED** [review]

**Comment by cincodenada** [comment]
Thanks for updating the title. One more detail: GitHub also doesn't recognize issue numbers in titles for whatever reason, so you have to put it in the body of the commit message to actually link the issues. I think if I just mention that this will fix #1850 in _this_ comment it should link things up though 🙂

File: locale/de_DE/LC_MESSAGES/django.po (hunk 2 above)
Comment (mouse-reeve) [reply locale/de_DE/LC_MESSAGES/django.po:295]: Good catch -- https://github.com/bookwyrm-social/bookwyrm/pull/1901

File: locale/de_DE/LC_MESSAGES/django.po [hunk 4]
```diff
@@ -17,62 +17,70 @@ msgstr ""
 "X-Crowdin-File: /[bookwyrm-social.bookwyrm] main/locale/en_US/LC_MESSAGES/django.po\n"
 "X-Crowdin-File-ID: 1553\n"
 
-#: bookwyrm/forms.py:365
+#: bookwyrm/forms.py:239
+msgid "Domain is blocked. Don't try this url again."
+msgstr "Die Domäne ist blockiert. Versuchen Sie diese Url nicht mehr."
```
Comment (mouse-reeve) [inline locale/de_DE/LC_MESSAGES/django.po:22]: Thank you so much for translating these! Would you be down to add them at https://translate.joinbookwyrm.com/ instead? That's the canonical source of translation strings, and I think it will overwrite changes you make locally like this. 

(I'd also be happy to copy it over if you would prefer that)

Comment (willhoh) [reply locale/de_DE/LC_MESSAGES/django.po:22]: I will gladly participate in the translation, but until I get to it, I would ask you to copy this translation in there.

**Comment by willhoh** [comment]
How do I have to do this correctly? Should I close the pull-request? Can then also the issue be closed?

**Comment by mouse-reeve** [comment]
Definitely don't close it! I'm going to check out this branch in my local instance and try out the changes, and let you know if anything needs changing. If not, I'll merge it and close the issue. I should be able to do that tomorrow.

**Comment by willhoh** [comment]
This pull request looks really bad. I will try to make it better next time.

I have one more comment to make: My pull request should also fix issue #1899.

**Comment by mouse-reeve** [comment]
You're welcome to open a new one if you prefer, but for what it's worth, it looks just fine to me.

**Comment by mouse-reeve** [comment]
This works great! I'm ready to merge once the tests are passing. I'm going to go ahead and suppress the line-too-long warnings

**Review by mouse-reeve — APPROVED** [review]

**Comment by willhoh** [comment]
> This works great! I'm ready to merge once the tests are passing. I'm going to go ahead and suppress the line-too-long warnings

I plan to fix this line-too-long issue or would ask @m1del to do so.

**Comment by mouse-reeve** [comment]
I think it's better to have the lines as-is -- I find that splitting up translation strings makes them hard to parse

**Comment by mouse-reeve** [comment]
Thank you for all your work on this, I know it's a lot to get in to a new codebase and I appreciate you sticking with iterating on changes. And I'm stoked to have this feature live 😄
